### ISSUE TITLE
Back-pressure on xadd qs

### ISSUE THREAD

**Issue by franz1981** [JCTools/JCTools#271]
Xadd qs can check if there are free chunks available before performing an offer: it would enable a weak forma back-pressure depending on the configured pooled chunks.

**Comment by nitsanw** [comment]
Is this to build some bounded capacity XADD queue? I need to dig deeper into the xadd queues before having this conversation I suspect :-)

**Comment by franz1981** [comment]
> Is this to build some bounded capacity XADD queue?
Yep altough is less precise then what we have on other CAS-based queues 
Just a note: performance of this ATM are BAD, really BAD XD

> I need to dig deeper into the xadd queues
That's my fault: I have changed them (especially mpmc) so many times that I haven't helped you to get a stable idea to work with :(
Anyway feel free to ping me on hangout and we can arrange a chat: I believe that in max 30 minutes of chat can cover both :+1:
